### DISCUSSION TITLE
Deeplinks for files

### DISCUSSION THREAD

**Discussion by ippocratis** [TechbeeAT/jtxBoard#898] [Ideas]
App already support attachments but is extremely limited to filesize by caldav design.

App already understands markdown. 

Maybe It would make sense to add markdown formated deeplinks for filepaths of files on device. 

I tried with files to the apps data dir `[some text](/data/user/0/at.techbee.jtx/files/example.pdf)` to avoid scoped storage limitations but also on a generic phone dir  `[some text](/storage/emulated/0/Documents/example.pdf)` without luck.

**Comment by patrickunterwegs** [comment]
Hi @ippocratis !
That's not as easy as it sounds in the first place: 

Apps cannot just access the filesystem of Android. Currently when an attachment is added to jtx Board, you select a specific file, the OS returns a content URI (that looks like content://...) and that allows the app to access the file for some time.
If an app wants to access ALL files without a user selecting it, it needs a special permission. This permission on the other side comes with certain limitations and additional policy requirements (see also https://developer.android.com/training/data-storage/manage-all-files).

Even when this is resolved, you would only have a link to a local file. So once you move the file the link would break, but even worse, when you sync the entry, the link would also not work anymore on another device.
If you only work on a single device, there is no file size limitation anyway. But for synchronised entries, storage access wouldn't really make a difference. 

A solution is managed attachments, but this also needs implementation on DAVx5 and probably also on the server side. You can check this discussion: #862

**Comment by bepolymathe** [comment]
Hi, 

Personally, I think the opposite would be practical. Have permanent links for tasks (with a unique ID) that we could integrate into Markdown note-taking software (logseq, joplin, obsidian, etc)... because it's always better to have the context of a task... and because the CALDAV standard will never be made for file management 🤔
